openedx/core/djangoapps/user_authn/migrations/0001_data__add_login_service.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.db import migrations
+
+
+def add_login_service(apps, schema_editor):
+    """
+    Adds a user and DOT application for the login service.
+    """
+    login_service_name = 'Login Service for JWT Cookies'
+    login_service_client_id = settings.JWT_AUTH['JWT_LOGIN_CLIENT_ID']
+    login_service_username = settings.JWT_AUTH['JWT_LOGIN_SERVICE_USERNAME']
+    login_service_email = login_service_username + '@fake.email'
+
+    # Get the User model using the AUTH_USER_MODEL settings since that is
+    # what the Application model expects at this time in the migration phase.
+    User = apps.get_model(settings.AUTH_USER_MODEL)
+    login_service_user, created = User.objects.get_or_create(
+        username=login_service_username,
+        email=login_service_email,
+    )
+    if created:
+        # Make sure the login service user's password is unusable.
+        # The set_unusable_password method is available on the other User model.
+        from django.contrib.auth.models import User
+        user = User.objects.get(username=login_service_username)
+        user.set_unusable_password()
+        user.save()
+
+    Application = apps.get_model(settings.OAUTH2_PROVIDER_APPLICATION_MODEL)
+    if not Application.objects.filter(client_id=login_service_client_id).exists():
```

Comment:
In theory, this could be the outermost if-statement, in case this exists with a different user and the setting was never set. No need to create another unused user.

Reply:
True.  I moved it up.